pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -520,6 +520,16 @@ func (host *BareMetalHost) WasProvisioned() bool {
 // WasExternallyProvisioned returns true when we think something else
 // is managing the image running on the host.
 func (host *BareMetalHost) WasExternallyProvisioned() bool {
+	// We do not want provisioned hosts to appear to be externally
+	// provisioned when we start deprovisioning them.
+	if host.Status.Provisioning.State == StateProvisioned {
+		return false
+	}
+	if host.Status.Provisioning.State == StateDeprovisioning {
+		return false
+	}
```

Comment:
As Russell observed, this doesn't help because once the server is deprovisioned it'll just drop back to "externally provisioned" again as soon as it hits "ready".